lib/Conversion/TorchToTcp/Elementwise.cpp

```diff
@@ -415,6 +415,86 @@ class ConvertAtenAtan2Op : public OpConversionPattern<AtenAtan2Op> {
   }
 };
 
+template <typename AtenOpT, int fillVal>
+class ConvertAtenZerosOnesPatternOp : public OpConversionPattern<AtenOpT> {
+public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+  LogicalResult
+  matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    auto outType = OpConversionPattern<AtenOpT>::getTypeConverter()
+                       ->convertType(op.getType())
+                       .template dyn_cast<RankedTensorType>();
+
+    if (!outType)
+      return rewriter.notifyMatchFailure(
+          op, "Only Ranked Tensor types are supported in TCP");
+
+    Type outElemTy = outType.getElementType();
+    if (!outElemTy.isIntOrFloat())
+      return rewriter.notifyMatchFailure(
+          op, "Output tensors must have integer or floating-point datatype");
+
+    if (!op.getLayout().getType().template isa<Torch::NoneType>())
+      return rewriter.notifyMatchFailure(op,
+                                         "Only default layout is supported");
+
+    bool pinMemory;
+    if (!op.getPinMemory().getType().template isa<Torch::NoneType>() &&
+        (!matchPattern(op.getPinMemory(), m_TorchConstantBool(&pinMemory)) ||
+         pinMemory)) {
+      return rewriter.notifyMatchFailure(
+          op, "Unsupported pin_memory, should be either None or false");
+    }
+
+    SmallVector<int64_t> shape;
+    if (!matchPattern(op.getSize(), m_TorchListOfConstantInts(shape))) {
```

Comment:
Ouch -- isn't TCP aiming to be more dynamic-shape aware?

Reply:
It looks like aten.zeros and aten.ones is used with static shape. Dynamic shape is supported in aten.zeros_like and aten.ones_like

Reply:
@zezhang I could be missing something here, but why can't we do what we're doing for zeros_like here (create a scalar constant & broadcast with dynamic dims)?  That should allow us to lift the restriction that `op.getSize()` is a constant.

Reply:
This must support dynamic shapes. I just had a discussion with @zezhang. He will be updating this and handling it the same way like `zeros_like`.

Reply:
Just updated.